room/src/Revature.Room.DataAccess/Repository.cs

```
@@ -0,0 +1,153 @@
+using Revature.Room.DataAccess.Entities;
+using Revature.Room.Lib;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Data = Revature.Room.DataAccess.Entities;
+
+namespace Revature.Room.DataAccess
+{
+  /// <summary>
+  /// Class in charge of methods that affect the state and data in the Room DB
+  /// </summary>
+  public class Repository : IRepository
+  {
+    private readonly RoomServiceContext _context;
+    private readonly IMapper _map;
+
+    public Repository(RoomServiceContext context, IMapper mapper)
+    {
+      _context = context;
+      _map = mapper;
+    }
+
+    /// <summary>
+    /// Method that creates a room
+    /// </summary>
+    /// <param name="myRoom"></param>
+    /// <returns></returns>
+    public async Task CreateRoomAsync(Lib.Room myRoom)
+    {
+      Data.Room roomEntity = _map.ParseRoom(myRoom);
+      await _context.AddAsync(roomEntity);
+    }
+
+    /// <summary>
+    /// Method that gets a room
+    /// </summary>
+    /// <param name="roomId"></param>
+    /// <returns></returns>
+    public async Task<List<Lib.Room>> ReadRoomAsync(Guid roomId)
+    {
+      //Find room by Guid and return that particular room
+      var listRoom = await _context.Room.Include(r => r.Gender).Include(r => r.RoomType).ToListAsync();
+
+      var x = listRoom.Where(r => r.RoomId == roomId).ToList();
+
+      return _map.ParseRooms(x).ToList();
+    }
+
+    /// <summary>
+    /// Method that updates the gender, lease start, end, and number of occupants of a room
+    /// </summary>
+    /// <param name="myRoom"></param>
+    /// <returns></returns>
+    public async Task UpdateRoomAsync(Lib.Room myRoom)
+    {
+      Data.Room roomEntity = await _context.Room.Where(r => r.RoomId == myRoom.RoomId)
+        .Include(r => r.Gender)
+        .Include(r => r.RoomType)
+        .FirstOrDefaultAsync() ?? throw new ArgumentNullException("There is no such room!");
```

if you want to throw an exception just use `FirstAsync`, that's what it's for. ArgumentNullException is wrong here because the argument is not null.